### PR TITLE
Added onAuthStateChanged to sdk reference docs

### PR DESCRIPTION
**Pull request by jrrglr**
Maybe it should not be at the bottom.

### CODE REVIEW

File: docs/reference/sdk/authentication.mdx [hunk 1]
```diff
@@ -115,3 +115,16 @@ Reset password for a user. This will send an email to the user with a link to re
...
+## `nhost.auth.onAuthStateChanged(event, session)`
+
+Use `onAuthStateChanged` to add a custom function that will trigger
+whenever the state of the user changed. Ex from signed in to signed out or
+vice versa.
+
+```js
+nhost.auth.onAuthStateChanged(({event, session}) => {
+  // event = 'SIGNED_IN' | 'SIGNED_OUT'
+  console.log(`auth state changed. State is not ${event} with session: ${session}`)
```
Comment (elitan) [inline docs/reference/sdk/authentication.mdx:128]: Some typos. I would update to:

```
  console.log(`Auth state changed. State is now ${event} with the session: ${session}`)
```

**Review by elitan — CHANGES_REQUESTED** [review]
Thanks for contributing. I added a short feedback.

**Comment by elitan** [comment]
Closed in favor of https://github.com/nhost/nhost/pull/132

**Comment by szilarddoro** [comment]
> Closed in favor of #132

This is the correct PR: https://github.com/nhost/nhost/pull/233

**Pull request closed without merging**